### PR TITLE
Main menu fps fix

### PR DESCRIPTION
**Pull request by NotStanley4330**
This is a fix for [Issue #488](https://github.com/exult/exult/issues/488) .  The fps button was not reading in a value from the exult.cfg file and instead simply defaulted to 5 on menu creation. Now it properly reads in the value and sets the button up accordingly.

### CODE REVIEW

**Comment by DominusExult** [comment]
CodeFactor complains about extra line in 214. Could you fix that?

**Comment by NotStanley4330** [comment]
> CodeFactor complains about extra line in 214. Could you fix that?

Fixed it!

File: gumps/GameEngineOptions_gump.cc [hunk 1]
```diff
@@ -180,6 +182,35 @@ void GameEngineOptions_gump::load_settings() {
...
+		frames = 5;
+		break;
+	case 8:
+		frames = 4;
+		break;
+	case 5:
+		frames = 3;
+		break;
+	case 6:
+		frames = 2;
```
Comment (rofl0r) [inline gumps/GameEngineOptions_gump.cc:201]: 6 has less fames than 5 ?
apart from that is seems the formula is frames = fps / 2 + fps % 2

Comment (NotStanley4330) [reply gumps/GameEngineOptions_gump.cc:201]: So oddly enough this is how it orders it. The fps button cycles in this order
{2,4,6,5,8,10}
I was mostly seeking to not change what was there too much, but I can completely redo it to have it make more sense

Comment (NotStanley4330) [reply gumps/GameEngineOptions_gump.cc:201]: this is the pre existing code

```
const int realframes = 1000 / gwin->get_std_delay();

	//this code generates the number of framerate options and their text
	//the functionality is kind of odd but I am leaving it alone.
	frames                        = -1;
	framerates[num_default_rates] = realframes;
	for (size_t i = 0; i < num_default_rates; i++) {
		if (realframes == framerates[i]) {
			frames = i;
			break;
		}
	}

	size_t num_framerates = num_default_rates;
	if (frames == -1) {
		num_framerates++;
		frames = num_default_rates;
	}
	frametext.clear();
	for (size_t i = 0; i < num_framerates; i++) {
		frametext.emplace_back(framestring(framerates[i]));
	}
```

Comment (NotStanley4330) [reply gumps/GameEngineOptions_gump.cc:201]: To be clear the ```frames``` value really just tells us the index of the value in the vector ```frametext``` to display on the button, and then that value is used to update the framerate when settings are saved.

Comment (NotStanley4330) [reply gumps/GameEngineOptions_gump.cc:201]: I do agree with it being a mess. I can look at just totally refactoring it today so it is so much more clear. It shouldn't break anything to just make it a predefined vector or enum. Best to just clean it up. Git blame tells me the code is 22 years old there so probably time to just fix it properly

Comment (marzojr) [reply gumps/GameEngineOptions_gump.cc:201]: I'd say go ahead and refactor it as much as you want.

Comment (NotStanley4330) [reply gumps/GameEngineOptions_gump.cc:201]: I'll refactor it this week and get it pushed back up!

Comment (NotStanley4330) [reply gumps/GameEngineOptions_gump.cc]: Refactored!

**Pull request merged**